src/main/java/org/micks/champmaker/auth/UserService.java

```diff
@@ -30,13 +31,33 @@ public void promoteToAdmin(String userId) {
 
     public void downgradeToPlayer(String userId) {
         log.info("Downgrade user {} to PLAYER", userId);
-        try(Connection connection = discGolfDbConnection.connect()) {
+        try (Connection connection = discGolfDbConnection.connect()) {
             PreparedStatement statement = connection.prepareStatement("UPDATE users SET role = 'PLAYER' WHERE user_id = ?");
-            statement.setString(1,userId);
+            statement.setString(1, userId);
             statement.execute();
             statement.close();
         } catch (SQLException e) {
             throw new RuntimeException("Error while downgrading user to PLAYER", e);
         }
     }
+
+    public UserRecord getUser(String userId) {
+        log.info("Getting user by Id: {}", userId);
+        try (Connection connection = discGolfDbConnection.connect()) {
+            PreparedStatement statement = connection.prepareStatement("SELECT user_id, username, role FROM users WHERE user_id = ?");
+            statement.setString(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                String dbUserId = resultSet.getString("user_id");
+                String username = resultSet.getString("username");
+                String roleString = resultSet.getString("role");
+                UserRole role = UserRole.valueOf(roleString.toUpperCase());
+                return new UserRecord(dbUserId, username, role);
+            } else {
+                return null;
```

Comment:
I would rather throw some exception than returning null. Exception will be more clear what actually happened and it should even never happen, if userId was taken from JWT token in a valid way.